File: scala_api/src/main/scala/application/models/SummaryOfMerics/ClassificationSummaryResult.scala
package application.models.SummaryOfMerics

import spray.json.{DefaultJsonProtocol, RootJsonFormat}

case class ClassificationSummaryResult(
  accuracy: Double = 0,
  error: Double = 0,
  precision: Double = 0,
  f1: Double = 0,
  weightedRecall: Double = 0,
  hammingLoss: Double = 0,
)

trait ClassificationSummaryResultJsonProtocol extends DefaultJsonProtocol {
  implicit val classSummaryResultFormat: RootJsonFormat[ClassificationSummaryResult] = jsonFormat6(ClassificationSummaryResult)
}

File: python_api/Models/ClassificationSummaryResult.py
class ClassificationSummaryResult:
    def __init__(self, accuracy=0, error=1, precision=0, f1=0, weightedRecall=0, hammingLoss=1):
        self.accuracy = accuracy
        self.error = error
        self.precision = precision
        self.f1 = f1
        self.weightedRecall = weightedRecall
        self.hammingLoss = hammingLoss
